fix: OPTIC-1835: Address issue affecting OCR region panel visualization

yyassi-heartex opened the pull request:
This pull request includes several updates to improve the user interface and functionality of the `OutlinerPanel` and `TextAreaRegionView` components. The most important changes include the reordering and addition of `RegionControls`, the introduction of a `canDelete` property, and various styling adjustments.

### OutlinerPanel improvements:

* [`web/libs/editor/src/components/SidePanels/OutlinerPanel/OutlinerTree.tsx`](diffhunk://#diff-f4db8bd107bcd19f6cfa67c1b88e801a4a5608c73b12b2bae96fea65b34bec08L424-L434): Reordered and re-added `RegionControls` to ensure proper rendering and interaction within the `OutlinerPanel`. [[1]](diffhunk://#diff-f4db8bd107bcd19f6cfa67c1b88e801a4a5608c73b12b2bae96fea65b34bec08L424-L434) [[2]](diffhunk://#diff-f4db8bd107bcd19f6cfa67c1b88e801a4a5608c73b12b2bae96fea65b34bec08R435-R445)
* [`web/libs/editor/src/components/SidePanels/OutlinerPanel/OutlinerTree.tsx`](diffhunk://#diff-f4db8bd107bcd19f6cfa67c1b88e801a4a5608c73b12b2bae96fea65b34bec08R616): Added `canDelete` property to `RegionItemDesc` to control the deletion capability of regions.

### TextAreaRegionView enhancements:

* [`web/libs/editor/src/tags/control/TextArea/TextAreaRegionView.jsx`](diffhunk://#diff-767aeedbca9fc500e5d03de72f91b52cd08c267505784cb4d4858d327e3c0ba0L20-R20): Introduced a `canDelete` property to control the visibility of the delete action for text areas. [[1]](diffhunk://#diff-767aeedbca9fc500e5d03de72f91b52cd08c267505784cb4d4858d327e3c0ba0L20-R20) [[2]](diffhunk://#diff-767aeedbca9fc500e5d03de72f91b52cd08c267505784cb4d4858d327e3c0ba0L78-R78) [[3]](diffhunk://#diff-767aeedbca9fc500e5d03de72f91b52cd08c267505784cb4d4858d327e3c0ba0L96-R96) [[4]](diffhunk://#diff-767aeedbca9fc500e5d03de72f91b52cd08c267505784cb4d4858d327e3c0ba0R136-R142) [[5]](diffhunk://#diff-767aeedbca9fc500e5d03de72f91b52cd08c267505784cb4d4858d327e3c0ba0R258)

### Styling adjustments:

* [`web/libs/editor/src/components/Comments/Comment/CommentFormButtons.scss`](diffhunk://#diff-29dbfcb79357da8f387ca4369a3947410aa71485cfc44c7e6b888255ae566cc0R5-R6): Adjusted button alignment within the comment form.
* [`web/libs/editor/src/components/SidePanels/OutlinerPanel/TreeView.scss`](diffhunk://#diff-b088514e9a8e8f1eaa65655520df7a3d197d0974369413cacbd5efce089f5253L54-R54): Updated padding and added text overflow handling for better visual consistency. [[1]](diffhunk://#diff-b088514e9a8e8f1eaa65655520df7a3d197d0974369413cacbd5efce089f5253L54-R54) [[2]](diffhunk://#diff-b088514e9a8e8f1eaa65655520df7a3d197d0974369413cacbd5efce089f5253R96-R97) [[3]](diffhunk://#diff-b088514e9a8e8f1eaa65655520df7a3d197d0974369413cacbd5efce089f5253L238)

yyassi-heartex committed:
fix: OPTIC-1835: Address issue affecting OCR region panel visualization

yyassi-heartex committed:
adding quick fix for comment form button alignment

yyassi-heartex committed:
lint cleanup

bmartel commented on web/libs/editor/src/components/Comments/Comment/CommentFormButtons.scss:
```diff
@@ -2,6 +2,8 @@
 .comment-form-buttons {
   &__buttons {
     display: flex;
+    height: 100%;
```
On the icons branch this might conflict, just as a heads up. The comment form and buttons had style adjustments there to fix a few visual issues that were observed.

yyassi-heartex replied:
great call out! no worries will resolve the conflict when we get there

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13970176336)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1835/ocr-region'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13970176336

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13970176336

yyassi-heartex committed:
cleaning up flex styling

yyassi-heartex committed:
further supporting smaller sizes

yyassi-heartex committed:
minor visual tweaks to handle smaller sizes

yyassi-heartex committed:
even smaller size

yyassi-heartex committed:
cleaning up layout so we leverage collapsed

yyassi-heartex committed:
lint cleanup
